sdk/src/auth/composable.ts

```diff
@@ -87,7 +86,7 @@ export const authentication = (mode: AuthenticationMode = 'cookie', config: Part
 		const refresh = async () => {
 			const awaitRefresh = async () => {
```

Comment:
A tad unrelated but I think we can omit the inner `awaitRefresh` wrapping here and directly return the request promise:
```ts
		const refresh = async () => {
			const authData = await storage.get();
			await resetStorage();

			const fetchOptions: RequestInit = {
				method: 'POST',
				headers: {
					'Content-Type': 'application/json',
				},
			};

			if ('credentials' in authConfig) {
				fetchOptions.credentials = authConfig.credentials;
			}

			const body: Record<string, string> = { mode };

			if (mode === 'json' && authData?.refresh_token) {
				body['refresh_token'] = authData.refresh_token;
			}

			fetchOptions.body = JSON.stringify(body);

			const requestUrl = getRequestUrl(client.url, '/auth/refresh');

			return request<AuthenticationData>(requestUrl.toString(), fetchOptions, client.globals.fetch).then((data) =>
				setCredentials(data).then(() => data),
			);
		};
```

Reply:
Let's merge the current solution first, as that solves the issue without other side effects. @br41nslug if you feel this is worth implementing as an improvement, lets do it in a separate PR 👍🏻

Reply:
Sure 👍

Reply:
This approach is missing the `refreshPromise` thats used to check if the SDK is currently doing a refresh call it needs to wait for.

Reply:
Hmm right 😅👌